### ISSUE TITLE
[DropdownBody]: focus lock takes and return lock for tooltip, causing unintended scrolling

### ISSUE THREAD

**Issue by jakobz** [epam/UUI#2005]
## Description

DropdownBody contains FocusLock feature, which allows putting focus in Dropdown when it gets open. It's a nice feature, and it works well in case when we use DropdownBody for dropdown menus, and similar things.

However, DropdownBody can also be used for tooltip-like UI, when Dropdown content is not interactive, and it gets open by hovering a target. Even our examples use this:
https://uui.epam.com/documents?id=dropdown&mode=doc&isSkin=null&category=components 

In case, DropdownBody also get's focus, and then returns it back, causing focus to return to the original element. This, in turn, causes the page to scroll back. This doesn't happen when we open Dropdown body on click, as when clicking the target, we set focus on it, and it's ok to return focus in this case.

Our demos uses focusLock=false option to avoid this. However, this is very error-prone, as it's not clear for users why they need it, it's easy to miss this setting, producing a tricky, and hard to stop bugs.

We discussed the issue, and there are two options to fix this:

a) Add ```focusLock``` prop to the ```IDropdownBodyProps```, so ```Dropdown``` can pass it to renderBody. Then, it's usually get's passed as is to the ```DropDownBody``` via ```{ ...props }```.

b) Add ```openedWith``` prop to the IDropdownBodyProps:
```
    /**
     * Indicates how dropdown was opened - by hovering or clicking
    */
    openedWith?: 'hover' | 'click';
```
In DropDownBody, default focusLock as ```props.openedWith !== 'hover'```

In both cases, we need to decide which value to pass to the new prop in the Dropdown component. I suggest to start with a simple detection, based on prop.openOnHover. This wouldn't handle cases when both openOnClick and openOnHover are true. However, they doesn't use at the same time in practice, and correct detection of how dropdown was opened would need serious adjustments of component's state management.

So I vote for option B, and a simple openedWith = openOnHover ? 'hover': 'click'. 

Also, please remove focusLock=false from our Dropdown Examples in scope of this feature, as it would become unnecessary and confusing after the fix.

**Comment by AlekseyManetov** [comment]
We've implemented the solution with preventing scroll to the focused component by focus return. 
https://github.com/epam/UUI/commit/ecb5440a83dc58598229846998d1fb000d642672

**Comment by NatalliaAlieva** [comment]
Released in 5.7.0 ver.
